apps/server/src/modules/files/repo/files.repo.spec.ts

```diff
@@ -0,0 +1,430 @@
+import { EntityManager, ObjectId } from '@mikro-orm/mongodb';
+import { Test, TestingModule } from '@nestjs/testing';
+import { MongoMemoryDatabaseModule } from '@shared/infra/database';
+import { StorageProviderEntity } from '@shared/domain';
+import { FileEntity } from '../entity';
+import { fileEntityFactory, filePermissionEntityFactory } from '../entity/testing';
+import { FilesRepo } from './files.repo';
+
+describe(FilesRepo.name, () => {
+	let repo: FilesRepo;
+	let em: EntityManager;
+	let module: TestingModule;
+
+	const mainUserId = new ObjectId().toHexString();
+	const otherUserId = new ObjectId().toHexString();
+
+	const otherUserFilesProps = [
+		// Test file created, owned and accessible only by the other user.
+		{
+			ownerId: otherUserId,
+			creatorId: otherUserId,
+			permissions: [filePermissionEntityFactory.build({ refId: otherUserId })],
+		},
+		// A second file also created, owned and accessible only by the other user.
+		{
+			ownerId: otherUserId,
+			creatorId: otherUserId,
+			permissions: [filePermissionEntityFactory.build({ refId: otherUserId })],
+		},
+	];
+
+	const setup = async () => {
```

Comment:
please move the setup methods into the respective describe blocks. this should also allow you to trim them a bit, as you dont have to support all the testcases at once.

Reply:
Good point, thanks :+1: .